waggle-dance-integration-tests/src/test/java/com/hotels/bdp/waggledance/WaggleDanceIntegrationTest.java

```diff
@@ -524,4 +527,25 @@ public void doesNotOverwriteConfigOnShutdownAfterAddingFederation() throws Excep
     assertThat(remoteMetastore.getMappedDatabases().get(0), is(REMOTE_DATABASE));
   }
 
+  @Test
+  public void restAPIGetStatus() throws Exception {
+    exit.expectSystemExitWithStatus(0);
+
+    runner = WaggleDanceRunner
+        .builder(configLocation)
+        .primary("primary", localServer.getThriftConnectionUri(), READ_ONLY)
+        .federate("waggle_remote", remoteServer.getThriftConnectionUri(), REMOTE_DATABASE)
+        .build();
+
+    runWaggleDance(runner);
+
+    RestTemplate rest = new RestTemplateBuilder().build();
+    PrimaryMetaStore primaryMetastore = rest.getForObject("http://localhost:18000/api/admin/federations/primary",
+        PrimaryMetaStore.class);
+    assertThat(primaryMetastore.getStatus(), is(MetaStoreStatus.AVAILABLE));
+    FederatedMetaStore federatedMetastore = rest
+        .getForObject("http://localhost:18000/api/admin/federations/waggle_remote", FederatedMetaStore.class);
+    assertThat(federatedMetastore.getStatus(), is(MetaStoreStatus.AVAILABLE));
+  }
+
```

Comment:
Is it worth having a test for UNAVAILABLE?

Reply:
Not here, golden path integeration test. The rest module has tests for the service for more specific stuff. This is just here to make sure all the stuff is wired up correctly.